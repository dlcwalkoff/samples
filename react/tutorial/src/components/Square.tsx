import React = require("react");

export interface SquareBoard {
    value: string,
    onClick: () => void
}

export class Square extends React.Component {
    constructor(public props: SquareBoard) {
        super(props);
    }

    render() {
        return (
            <button className='square' onClick={() => this.props.onClick()}>
                {this.props.value}
            </button>
        );
    }
}