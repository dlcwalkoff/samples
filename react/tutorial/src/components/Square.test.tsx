import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Square } from './Square';
import React = require('react');


describe('Square', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Square value='X' onClick={() => {}} />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should handle the click event', () => {
    const logText = 'Square handle click test';
    console.log = jest.fn();

    const output = shallow(
      <Square value='X' onClick={() => console.log(logText)} />
    );

    output.simulate('click');
    expect(console.log).toHaveBeenCalledWith(logText);
  });
});