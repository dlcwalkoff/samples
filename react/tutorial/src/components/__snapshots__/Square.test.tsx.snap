// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render correctly 1`] = `
<button
  className="square"
  onClick={[Function]}
>
  X
</button>
`;
